app/services/send_add_email_confirmation.rb

```diff
@@ -58,7 +59,8 @@ def send_email_associated_with_another_account_email
 
   def send_confirmation_email
     UserMailer.with(user: user, email_address: email_address).add_email(
-      confirmation_token,
+      token: confirmation_token,
+      request_id:,
```

Comment:
Can we add some test coverage in `send_add_email_confirmation_spec.rb` that the mailer is sent with the `request_id` ?

Reply:
> Where was it going? I'm wondering if this could be related to https://github.com/18F/identity-idp/pull/11439#discussion_r1825703769 about using StoreSpMetadataInSession to ensure that the session is initialized with the request_id.

One thing happening is I added `return sign_up_select_email_url if sp_session.key?(:request_id)` here instead of using the url parameter
https://github.com/18F/identity-idp/blob/bed2a55da0f156d7226f1e871f7c9ee4dcbe54ac/app/controllers/application_controller.rb#L238
And depending on if it's before or after `return sp_session_request_url_with_updated_params if sp_session.key?(:request_url)` selects if it's going to send the user to the select email view or confirm to share view which is not good depending on which you expect to land on.

I tried adding a version of `store_sp_metadata_in_session` at the emails_controller:add but session[:sp] doesn't persist when you open the confirmation link in a private window.